File: backend/app/schemas/note.py
"""Pydantic schemas for Employee Note validation."""

from pydantic import BaseModel, Field
from typing import Optional


class NoteCreate(BaseModel):
    employee_id: str
    note: str = Field(..., min_length=1)
    created_by: str = Field(..., min_length=1, max_length=255)


class NoteUpdate(BaseModel):
    note: Optional[str] = Field(None, min_length=1)
    created_by: Optional[str] = Field(None, min_length=1, max_length=255)

File: frontend/types/note.ts
export interface EmployeeNote {
  id: string;
  employee_id: string;
  note: string;
  created_by: string;
  created_at: string;
}

export interface NoteCreate {
  employee_id: string;
  note: string;
  created_by: string;
}

export interface NoteUpdate {
  note?: string;
  created_by?: string;
}
